x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java

```diff
@@ -504,7 +509,29 @@ private void assertIndexReplicaShardsAreNotAllocated(String indexName) {
     }
 
     private void assertIndexSetting(String index, String setting, String expectedValue) {
-        var response = client().admin().indices().getSettings(new GetSettingsRequest().indices(index)).actionGet();
+        var response = client().admin().indices().prepareGetSettings(index).get();
         assertThat(response.getSetting(index, setting), equalTo(expectedValue));
     }
+
+    private static class BackgroundAssertion extends Thread {
+
+        private final Runnable assertion;
+        private final AtomicBoolean stopped = new AtomicBoolean(false);
+
+        BackgroundAssertion(Runnable assertion) {
+            this.assertion = assertion;
+            start();
+        }
+
+        @Override
+        public void run() {
+            while (stopped.get() == false) {
+                assertion.run();
+            }
+        }
+
+        public void verify() {
```

Comment:
I suppose this should be named `stop`? I think we should remove this background thread and verify explicitly the points we are interested in.

Also, when stopping a thread it is good to `join` too - to ensure it does not affect any subsequent tests - and we do not take down the cluster while the assertion is running.